File: examples/typescript/basic.ts
import { pathToFileURL } from "node:url";

import {
  createIdentifold,
  createNamespaceRegistry,
  parsePublicId,
} from "@greyfoundry/identifold";

export async function buildIdentity() {
  const registry = createNamespaceRegistry([{ publicPrefix: "order" }]);
  const identifiers = createIdentifold({ registry });
  const identity = await identifiers.create("order");
  const parsed = parsePublicId(identity.pid, "order");

  return {
    mid: identity.mid,
    namespace: parsed.namespace,
    pid: identity.pid,
    roundTrip: parsed.machineId === identity.mid,
  };
}

if (
  process.argv[1] !== undefined &&
  import.meta.url === pathToFileURL(process.argv[1]).href
) {
  console.log(JSON.stringify(await buildIdentity(), null, 2));
}

File: examples/python/basic.py
from __future__ import annotations

import json
from typing import TypedDict

from identifold import (
    create_machine_id,
    parse_public_id,
    public_id_from_machine_id,
)


class ExampleIdentity(TypedDict):
    mid: str
    namespace: str
    pid: str
    roundTrip: bool


def build_identity() -> ExampleIdentity:
    machine_id = create_machine_id()
    public_id = public_id_from_machine_id(machine_id, "order")
    parsed = parse_public_id(public_id, "order")

    return {
        "mid": machine_id,
        "namespace": parsed.namespace,
        "pid": public_id,
        "roundTrip": parsed.machine_id == machine_id,
    }


if __name__ == "__main__":
    print(json.dumps(build_identity(), indent=2))
